searx: 0.17.0 -> 0.18.0

illustris opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Searx 0.17 doesn't seem to work any more. Queries don't return results.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/servers/web-apps/searx/default.nix:
```diff
@@ -4,20 +4,26 @@ with python3Packages;
     sed -i 's/==.*$//' requirements.txt
   '';
 
+  SEARX_DEBUG = "true";
```
Can you export this in the pre or post phase where it is required?

illustris replied:
I'll verify it builds that way and commit.

SuperSandro2000 commented on pkgs/servers/web-apps/searx/default.nix:
```diff
@@ -31,7 +37,7 @@ toPythonModule (buildPythonApplication rec {
 
   preCheck = ''
-    rm tests/test_robot.py # A variable that is imported is commented out
+    #rm tests/test_robot.py # A variable that is imported is commented out
```
Please remove this if it is no longer required.

illustris replied:
Whoops... missed that in the cleanup.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109114` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>searx</li>
</ul>
</details>

illustris commented:
Requested changes made.

rnhmjoj commented:
This update broke the uWSGI support in the NixOS module.
I see a `ValueError: Empty module name`, but nothing else. I'm currently bisecting.

rnhmjoj commented:
Found it:
```
116f7a6daaa2d29db30bc0765427d011fa99ce9c is the first bad commit
commit 116f7a6daaa2d29db30bc0765427d011fa99ce9c
Author: Noémi Ványi <sitbackandwait@gmail.com>
Date:   Fri Oct 9 15:49:48 2020 +0200

    Force admins to set secret_key if debug mode is disabled
    
    This commit also enables debug mode for unit tests.

```
Looks like we must set `secret_key` in the test.

rnhmjoj committed:
nixos/searx: fix for 0.18.0 update

rnhmjoj committed:
nixos/searx: remove settings.yml merging script

Starting with 0.18.0, searx can merge the settings.yml automatically
with the default values, so the script is obsolete.

rnhmjoj commented on pkgs/servers/web-apps/searx/default.nix:
```diff
@@ -4,20 +4,28 @@ with python3Packages;
+  ];
+
   postPatch = ''
     sed -i 's/==.*$//' requirements.txt
```
Instead of adding a patch, you can use `sed` here to simply delete the test:
```
    # skip failing test
    sed -i '/test_json_serial(/,+3d' tests/unit/test_standalone_searx.py
```

SuperSandro2000 commented:
@rnhmjoj maybe we could add you as a maintainer so that you get auto notified of updates. Should be a different PR probably.

rnhmjoj commented:
@SuperSandro2000 yes, I'm pretty involved in searx at this point, it would make sense.

@Illustris, thank you! Everything looks good, can you just squash the fix commits into the first one? You can use `git rebase -i HEAD~5`,

illustris committed:
searx: 0.17.0 -> 0.18.0

illustris commented:
@rnhmjoj I just notices that I squashed your two commits commits to `nixos/modules/services/networking/searx.nix` and `nixos/tests/searx.nix`. I've rebased to include just the three commits to `pkgs/servers/web-apps/searx/default.nix`. Please add the service and test changes again.

rnhmjoj commented:
@illustris Done, and thank you for contributing!
